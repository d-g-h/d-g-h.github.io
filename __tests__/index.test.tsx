/**
 * @jest-environment jsdom
 */
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import Time from "../components/time";

describe("Time", () => {
  it("renders start date to present", () => {
    render(<Time start={"2024-02"} />);

    const start = screen.getByText("February 2024");
    const end = screen.getByText("- present");

    expect(start).toBeInTheDocument();
    expect(end).toBeInTheDocument();
  });

  it("renders start date to end date", () => {
    render(<Time start={"2022-02"} end={"2022-12"} />);

    const start = screen.getByText("February 2022");
    const hypen = screen.getByText("-");
    const end = screen.getByText("December 2022");

    expect(start).toBeInTheDocument();
    expect(hypen).toBeInTheDocument();
    expect(end).toBeInTheDocument();
  });

  it("renders html correctly", () => {
    const tree = renderer.create(<Time start={"2024-04"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
