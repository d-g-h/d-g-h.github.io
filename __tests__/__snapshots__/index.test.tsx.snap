// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time renders html correctly 1`] = `
[
  <time
    dateTime="2024-04"
  >
    April 2024
  </time>,
  "- present",
]
`;
